environ/src/environ/core.cljc

```diff
@@ -0,0 +1,55 @@
+(ns environ.core
+  #?(:clj
+     (:require [clojure.edn :as edn]
+               [clojure.java.io :as io]))
+  (:require [clojure.string :as str])
+  #?(:cljs
+     (:require-macros [cljs.core :refer [exists?]])))
+
+(defn- keywordize [s]
+  (-> (str/lower-case s)
+      (str/replace "_" "-")
+      (str/replace "." "-")
+      (keyword)))
+
+(defn- sanitize-key [k]
+  (let [s (keywordize (name k))]
+    (if-not (= k s) (println "Warning: environ key" k "has been corrected to" s))
+    s))
+
+(defn- sanitize-val [k v]
+  (if (string? v)
+    v
+    (do (println "Warning: environ value" (pr-str v) "for key" k "has been cast to string")
+        (str v))))
+
+(defn- read-system-env []
```

Comment:
In all of these functions, the reader conditionals could be placed at the top level.

So something more like:

```clojure
(defn- keywordize-keys [m]
  (->> m (map (fn [[k v]] [(keywordize k) v])) (into {})))

#?(:cljs
   (defn- read-process-env []
     (when (exists? js/process)
       (keywordize-keys (js->cljs js/process.env))))))

#?(:clj
  (defn- read-system-env []
    (keywordize-keys (System/getenv))))

#?(:clj
  (defn- read-system-props []
    (keywordize-keys (System/getProperties))))

#?(:clj
   (defn- read-env-file [f]
     (when-let [env-file (io/file f)]
       (when (.exists env-file)
         (into {} (for [[k v] (edn/read-string (slurp env-file))]
                    [(sanitize-key k) (sanitize-val k v)]))))))

(def ^{:doc "A map of environment variables."}
  env
  #?(:clj
     (merge
      (read-env-file ".lein-env")
      (read-env-file (io/resource ".boot-env"))
      (read-system-env))
     :cljs
     (read-process-env)))